NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -514,7 +514,35 @@ public StateQuery()
                     }
 
                     return null;
-                });
+            });
+            Field<CombinationSlotStateType>(
+                "CombinationSlot",
+                description: "Allows you to see crafting slot data.",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<AddressType>>
+                    {
+                        Name = "avatarAddress",
+                        Description = "Address of avatar."
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "slot",
+                        Description = "Slot index 0-3"
+                    }
+                ),
+                resolve: context =>
+                {
+                    var agentAddress = context.GetArgument<Address>("avatarAddress");
+                    var index = context.GetArgument<int>("slot");
```

Comment:
Since `index` is only valid when between 0 and 3, checking and throwing an error against illegal values might be nice. 😄

Reply:
I'll add that 👍

Reply:
added it now, sorry for delay. Had my system drive fail so had to install the whole OS again.